python/pyspark/errors/exceptions/captured.py

```diff
@@ -379,5 +379,13 @@ def fragment(self) -> str:
     def callSite(self) -> str:
         return str(self._q.callSite())
 
+    def pysparkFragment(self) -> Optional[str]:  # type: ignore[return]
+        if self.contextType() == QueryContextType.DataFrame:
+            return str(self._q.pysparkFragment())
+
+    def pysparkCallSite(self) -> Optional[str]:  # type: ignore[return]
```

Comment:
how are these two different? The given example only shows the call site
```
== PySpark call site ==
"divide" was called from
/.../spark/python/test_pyspark_error.py:8
```

Reply:
oh, `divide` is the fragement?

Reply:
By any chance could we comment what "fragment" and "callSite" are at PySparkCurrentOrigin?

Reply:
> how are these two different? The given example only shows the call site
> 
> ```
> == PySpark call site ==
> "divide" was called from
> /.../spark/python/test_pyspark_error.py:8
> ```

Yes, the `fragment` is a function identifier and the call site is the file name including line number.

Reply:
> By any chance could we comment what "fragment" and "callSite" are at PySparkCurrentOrigin?

Sounds good. Let me add a comment

Reply:
Thank you!